pkg/brokerapi/openservicebroker/util/fake_broker_server.go

```diff
@@ -0,0 +1,142 @@
+	router.HandleFunc("/v2/service_instances/{id}", f.deprovisionHandler).Methods("DELETE")
+	go http.ListenAndServe(":"+strconv.Itoa(port), router)
+}
+
+// SetResponseStatus sets the default response status of the broker to the given HTTP status code
+func (f *FakeBrokerServer) SetResponseStatus(status int) {
+	f.responseStatus = status
+}
+
+// SetAsynchronous sets the number of polls before finished, final state, and operation for asynchronous operations
+func (f *FakeBrokerServer) SetAsynchronous(numPolls int, shouldSucceed bool, operation string) {
+	f.pollsRemaining = numPolls
+	f.shouldSucceedAsync = shouldSucceed
+	f.operation = operation
+}
+
+// HANDLERS
+
+func (f *FakeBrokerServer) catalogHandler(w http.ResponseWriter, r *http.Request) {
+	util.WriteResponse(w, http.StatusOK, &brokerapi.Catalog{})
+}
+
+func (f *FakeBrokerServer) lastOperationHandler(w http.ResponseWriter, r *http.Request) {
+	req := &brokerapi.LastOperationRequest{}
+	if err := util.BodyToObject(r, req); err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	var state string
+	switch {
+	case f.pollsRemaining > 0:
+		f.pollsRemaining--
+		state = "in progress"
+	case f.shouldSucceedAsync:
+		state = "succeeded"
+	default:
+		state = "failed"
+	}
+
+	resp := brokerapi.LastOperation{
+		State: state,
+	}
+	util.WriteResponse(w, http.StatusOK, &resp)
+}
+
+func (f *FakeBrokerServer) provisionHandler(w http.ResponseWriter, r *http.Request) {
+	req := &brokerapi.CreateServiceInstanceRequest{}
+	if err := util.BodyToObject(r, req); err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	if !req.AcceptsIncomplete {
+		// Synchronous
+		util.WriteResponse(w, f.responseStatus, &brokerapi.CreateServiceInstanceResponse{})
+	} else {
+		// Asynchronous
+		resp := brokerapi.CreateServiceInstanceResponse{
+			Operation: f.operation,
+		}
+		util.WriteResponse(w, http.StatusAccepted, &resp)
+	}
+}
+
+func (f *FakeBrokerServer) deprovisionHandler(w http.ResponseWriter, r *http.Request) {
+	req := &brokerapi.DeleteServiceInstanceRequest{}
+	if err := util.BodyToObject(r, req); err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	if !req.AcceptsIncomplete {
+		// Synchronous
+		util.WriteResponse(w, f.responseStatus, &brokerapi.DeleteServiceInstanceResponse{})
+	} else {
+		// Asynchronous
+		resp := brokerapi.CreateServiceInstanceResponse{
+			Operation: f.operation,
+		}
+		util.WriteResponse(w, http.StatusAccepted, &resp)
+	}
+}
+
+func (f *FakeBrokerServer) updateHandler(w http.ResponseWriter, r *http.Request) {
+	// TODO: Implement
+	util.WriteResponse(w, http.StatusForbidden, nil)
+}
+
+func (f *FakeBrokerServer) bindHandler(w http.ResponseWriter, r *http.Request) {
+	// TODO: Implement
+	util.WriteResponse(w, http.StatusForbidden, nil)
+}
+
+func (f *FakeBrokerServer) unbindHandler(w http.ResponseWriter, r *http.Request) {
+	// TODO: Implement
```

Comment:
Well it seems like these should already be captured by the issues to implement update, bind and unbind (well maybe not bind :))